### ISSUE TITLE
Dependency conflict on Ubuntu 18

### ISSUE THREAD

**Issue by trondhindenes** [PowerShell/PowerShell#6588]
Steps to reproduce
------------------
Attempted install of deb package for ubuntu 17 on Ubuntu 18.04
output of `lsb_release -a`:
```
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu Bionic Beaver (development branch)
Release:	18.04
Codename:	bionic
```

Not sure what I expected, but Powershell seems to rely on `libcurl3`. Ubuntu 18.04 uses `libcurl4`, which seems to conflict with libcurl3. This makes it impossible to have Powershell and the bundled version of curl etc installed at the same time.

It would be awesome if a deb package compatible with Ubuntu 18 was released.

**Comment by fvanroie** [comment]
I got PS Core 6.0.1 for 17.10 [working on Bionic Beaver Beta](https://github.com/fvanroie/PS_OPNsense/wiki/Install-PowerShell-on-Ubuntu-18.04-beta).

**Comment by trondhindenes** [comment]
Yes, but by downgrading libcurl you likely broke it for other tools like curl etc.
For some packages one can work around the problem by manually editing the dependencies in the deb:
```
$ wget https://downloads.slack-edge.com/linux_releases/slack-desktop-3.1.0-amd64.deb
$ dpkg-deb -R slack-desktop-3.0.1-amd64.deb slack
$ vim slack/DEBIAN/control # and replace libcurl3 with libcurl4
$ dpkg-deb -b slack slack.deb
$ sudo dpkg -i slack.deb
```

**Comment by trondhindenes** [comment]
By doing the same as above for Powershell, and editing the dependencies its clear that the powershell deb package directly depends on libcurl3 and an incompatible version of libicu.

By changing these two and repackaging the deb, I was able to install the powershell package just fine without downgrading to libcurl3.

I completely realize that more testing is needed, but as long as Powershell directly depends on libcurl and libicu, @iSazonov 's comment seems to be incorrect.

**Comment by markekraus** [comment]
as a work around, couldn't you install the libcurl3 and libicu to something like `/opt/powershelllib/` and then launch PowerShell with `LD_LIBRARY_PATH=/opt/powershelllib pwsh`?

**Comment by kenk667** [comment]
I was able to successfully install on 18.04 by changing the dependencies for https://github.com/PowerShell/PowerShell/releases/download/v6.1.0-preview.2/powershell_6.1.0-preview.2-1.ubuntu.17.04_amd64.deb

```bash
dpkg-deb --raw-exatrct powershell_6.1.0-preview.2-1.ubuntu.17.04_amd64.deb 6.1.0_patched
```

Then alter two patterns, curl and libicu dependencies to current versions;

```bash
sudo awk '{gsub("curl3","curl4");gsub("libicu57","libicu60");print}' 6.1.0_pathched/DEBIAN/control
```

Repackage as a debian package;

```bash
dpkg-deb --build 6.1.0_pathched/ powershell_6.1.0-patched.2-1.ubuntu.18.04_amd64.deb
```

Now just install the newly patched .deb package, I like to use gdebi to install;

```bash
sudo gdebi powershell_6.1.0-patched.2-1.ubuntu.18.04_amd64.deb
```

I haven't estensively tested this, but seems to be functional and stable so far, but maybe a viable fix for 18.04 install.

**Comment by trondhindenes** [comment]
Same here, not seeing any problems. .Net core doesn't seem to play into it at all.
